index.js

```diff
@@ -24,23 +23,17 @@ getProjects(function (err, projects) {
   }
 })
 
-controller.on('rtm_open', function (bot, message) {
-  getUsers(bot)
-  console.log(projectList[0])
+getUsers(function (err, users) {
+  if (err) console.log(err + ' c')
+  else {
+    console.log('users retrieved ok')
+  }
 })
 
-/* let getUser = function (bot, user, callback) {
-  controller.storage.users.get(user, function(err, user_data) {
-    if (err) {
-      console.log(err)
-      getUsers(bot)
-    }
-    callback(null, user)
-  })
-} */
-/* getUser(bot, 'dara', function(err, dara) {
-  console
-}) */
+// connect the bot to a stream of messages
+controller.spawn({
```

Comment:
Currently the logic is like this:

getProjects()
getUsers()
controller.spawn()

It is incorrect to assume these functions complete sequentially because they're asynchronous. `controller.spawn` could complete before the other two functions finish.

Perhaps we could do something like:

```js
getProjects(function(err, projects) {
  // error check
  // save projects
  getUsers(function(err, users) {
    // error check
    // save users
    controller.spawn({token: config.slackToken}).startRTM()
  })
})
```